src/bug-filing/common/issue-filing-url-string-utils.ts

```diff
@@ -1,60 +1,50 @@
-    getFooter: (info: EnvironmentInfo) => string;
-    collapseConsecutiveSpaces: (input: string) => string;
-    formatAsMarkdownCodeBlock: (input: string) => string;
     getSelectorLastPart: (selector: string) => string;
     standardizeTags: (data: CreateIssueDetailsTextData) => string[];
     appendSuffixToUrl: (url: string, suffix: string) => string;
 };
 
-export const IssueFilingUrlStringUtils: IssueUrlCreationUtils = {
-    getTitle: (data: CreateIssueDetailsTextData): string => {
-        const standardTags = IssueFilingUrlStringUtils.standardizeTags(data);
+export const createUtils = (): IssueUrlCreationUtils => {
+    const getTitle = (data: CreateIssueDetailsTextData): string => {
+        const standardTags = standardizeTags(data);
         let prefix = standardTags.join(',');
         if (prefix.length > 0) {
             prefix = prefix + ': ';
         }
 
-        const selectorLastPart = IssueFilingUrlStringUtils.getSelectorLastPart(data.ruleResult.selector);
+        const selectorLastPart = getSelectorLastPart(data.ruleResult.selector);
 
         return `${prefix}${data.ruleResult.help} (${selectorLastPart})`;
-    },
-    getFooter: (environmentInfo: EnvironmentInfo): string => {
-        return (
-            `This accessibility issue was found using ${title} ` +
-            `${environmentInfo.extensionVersion} (axe-core ${environmentInfo.axeCoreVersion}), ` +
-            'a tool that helps find and fix accessibility issues. Get more information & download ' +
-            'this tool at http://aka.ms/AccessibilityInsights.'
-        );
-    },
-    collapseConsecutiveSpaces: (input: string): string => input.replace(/\s+/g, ' '),
-    formatAsMarkdownCodeBlock: (input: string): string =>
-        input
-            .split('\n')
-            .map(line => `    ${line}`)
-            .join('\n'),
-    getSelectorLastPart: (selector: string): string => {
+    };
+    const getSelectorLastPart = (selector: string): string => {
         let selectorLastPart = selector;
         if (selector.lastIndexOf(' > ') > 0) {
             selectorLastPart = selector.substr(selector.lastIndexOf(' > ') + 3);
         }
         return selectorLastPart;
-    },
-    appendSuffixToUrl: (url: string, suffix: string): string => {
+    };
+    const appendSuffixToUrl = (url: string, suffix: string): string => {
         if (endsWith(url, suffix) || endsWith(url, `${suffix}/`)) {
             return url;
         }
 
         return `${url}/${suffix}/`;
-    },
-    standardizeTags: (data: CreateIssueDetailsTextData): string[] => data.ruleResult.guidanceLinks.map(tag => tag.text.toUpperCase()),
+    };
+    const standardizeTags = (data: CreateIssueDetailsTextData): string[] =>
+        data.ruleResult.guidanceLinks.map(tag => tag.text.toUpperCase());
+
+    return {
+        getTitle,
+        getSelectorLastPart,
+        appendSuffixToUrl,
+        standardizeTags,
+    };
 };
+
+export const IssueFilingUrlStringUtils = createUtils();
```

Comment:
What does this indirection add vs just exporting the object returned by createUtils directly?

Reply:
It makes it easier to reuse functions inside the object being created. Look at `getTitle` for example, it use `standardizeTags`. For me it reads better like this instead of `IssueFilingUrlStringUtils.standardizeTag`

Reply:
I agree it reads a little better, but I don't think it's enough better to be worth having to type the names multiple times and losing F12/codelens.

Reply:
Not sure if I follow you're F12/codelens comment. That's around IDE not about the code itself so it feels more subjective than not. I'm still updating to not expose the function just the result and looking into actually removing the function itself (not sure if I actually do that last part though)

Reply:
Updated. Does this work better?